networking_aci/plugins/ml2/drivers/mech_aci/driver.py

```diff
@@ -300,6 +307,37 @@ def on_fabric_transit_created(self, resource, event, trigger, payload):
         # send to agent
         self.rpc_notifier.sync_network(sync_data)
 
+    @registry.receives(aci_const.CC_FABRIC_NET_GW, [events.BEFORE_UPDATE])
+    def on_network_gateway_move(self, resource, event, trigger, payload):
+        network_id = payload.metadata['network_id']
+        if not payload.metadata['move-to-cc-fabric']:
+            LOG.warning("Moving a gateway _away_ from cc-fabric is not supported by ACI yet (for network %s)",
+                        network_id)
+            return
+
+        LOG.info("Got request to move l3 gateway away from ACI for network %s", network_id)
+
+        try:
+            network = self._plugin.get_network(payload.context, network_id)
+        except n_exc.NetworkNotFound as e:
+            LOG.error("Could not find network %s - network does not exist! (Error was %s)",
+                      network_id, e)
+            return
+
+        if not network[extnet_def.EXTERNAL]:
+            LOG.error("Got event to move gateway of network %s, which is NOT an external network!", network_id)
+            return
+
+        for n, subnet_id in enumerate(network['subnets']):
+            subnet = self.get_subnet(payload.context, subnet_id)
+            subnetpool_id = subnet['subnetpool_id']
+            address_scope_name = self.db.get_address_scope_name(context._plugin_context, subnetpool_id)
+            if address_scope_name is None:
+                continue
+            last_on_network = n + 1 == len(network['subnets'])
+            self.rpc_notifier.delete_subnet(subnet, external=True, address_scope_name=address_scope_name,
+                                            last_on_network=last_on_network)
```

Comment:
Is it important that we might not have sent out a delete for subnets without address_scope_name and thus `last_on_network` is maybe not actually the last on the network, just the last in the list of subnets?

Reply:
That's actually a good point. I'm not 100% sure on the whole "last_network" semantic, I think in the end it'll delete the l3out and stuff and I'm not sure that this is something the syncloop will properly do.... So, I think I'll do an extra iteration over it. :)